Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/FeedBack.hs

```diff
@@ -0,0 +1,46 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.FeedBack where
+
+import Domain.Types.Ride as Ride
+import Kernel.Prelude
+import Kernel.Storage.Esqueleto
+import Kernel.Tools.Metrics.CoreMetrics
+import Kernel.Types.APISuccess
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import qualified SharedLogic.CallBAP as CallBAP
+import Storage.CachedQueries.CacheConfig
+import qualified Storage.Queries.Booking as QBooking
+import qualified Storage.Queries.Ride as QRide
+import Tools.Error
+
+data FeedbackReq = FeedbackReq
+  { rideId :: Id Ride,
+    ratingValue :: Int,
+    feedbackDetails :: Maybe Text
+  }
+  deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
+
+feedback :: (EncFlow m r, CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, CoreMetrics m, HasFlowEnv m r '["nwAddress" ::: BaseUrl], HasHttpClientOptions r c, HasShortDurationRetryCfg r c) => FeedbackReq -> m APISuccess
+feedback request = do
+  unless (request.ratingValue `elem` [1 .. 5]) $ throwError InvalidRatingValue
+  let rideId = request.rideId
+  ride <- QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  unless (ride.status == COMPLETED) $ throwError (RideInvalidStatus "Feedback available only for completed rides.")
```

Comment:
I think it should be possible for CANCELLED too?

Reply:
for now we are only doing it for COMPLETED Status . cc :: @gouri-muley